client/engine/engine.go

```diff
@@ -44,6 +44,7 @@ type APIEvent struct {
 type ObjectiveChangeEvent struct {
 	// These are objectives that are now completed
 	CompletedObjectives []protocols.Objective
+	InvalidObjectiveIds []protocols.ObjectiveId
```

Comment:
I wonder if we should be returning an `error` alongside the `ObjectiveId`? There are probably some kinds of errors that a user could deal with (ie: add more funding to solve an out of funds error), so it could be useful to communicate that back.